src/sidebar/groups.js

```diff
@@ -16,7 +16,7 @@ var STORAGE_KEY = 'hypothesis.groups.focus';
 var events = require('./events');
 
 // @ngInject
-function groups(localStorage, serviceUrl, session, $rootScope, $http) {
+function groups(localStorage, serviceUrl, session, $rootScope, store) {
```

Comment:
Would I be correct in guessing that Angular’s dependency injection means renaming this parameter is enough for some other bit of code somewhere else to know that it needs to pass in a different object?

Reply:
You are indeed correct. In `src/sidebar/app.js` there are a series of calls which associate string names (eg. "store") with functions that implement the service or UI element.

When Angular instantiates a service it looks at the names of the input parameters and instantiates each dependency first and passes it in as an argument in the appropriate slot. The `// @ngInject` comment above is a hint to a compile-time preprocessor to save the names of the dependencies as an array in this file when minifying/compressing the code, which would otherwise replace them with shorter names to save space.